File: SAE6/jardin_de_cocagne/lib/models/basket_product.dart
// lib/models/basket_product.dart
class BasketProduct {
  final int? id;
  final int? productId;
  final String? productName;
  final double quantity;
  final String? unit;
  final String? category;

  BasketProduct({
    this.id,
    this.productId,
    this.productName,
    required this.quantity,
    this.unit,
    this.category,
  });

  factory BasketProduct.fromJson(Map<String, dynamic> json) {
    return BasketProduct(
      id: json['id'],
      productId: json['product_id'],
      productName: json['product_name'],
      quantity: double.parse(json['quantity'].toString()),
      unit: json['unit'],
      category: json['category'],
    );
  }
}

File: SAE6/jardin_de_cocagne_docker/api/src/models/BasketProduct.ts
export interface BasketProduct {
    id: number;
    basket_id: number;
    product_id: number;
    quantity: number;
    product_name?: string;
    category?: string;
    unit?: string;
    created_at?: Date;
    updated_at?: Date;
  }
  
  export interface CreateBasketProductDTO {
    basket_id: number;
    product_id: number;
    quantity: number;
  }
  
  export interface UpdateBasketProductDTO {
    quantity?: number;
  }